src/wallet/walletdb.cpp

```diff
@@ -1499,17 +1499,19 @@ std::unique_ptr<WalletDatabase> MakeDatabase(const fs::path& path, const Databas
     if (format == DatabaseFormat::SQLITE) {
 #ifdef USE_SQLITE
         return MakeSQLiteDatabase(path, options, status, error);
-#endif
+#else
         error = Untranslated(strprintf("Failed to open database path '%s'. Build does not support SQLite database format.", fs::PathToString(path)));
         status = DatabaseStatus::FAILED_BAD_FORMAT;
         return nullptr;
+#endif
     }
 
 #ifdef USE_BDB
     return MakeBerkeleyDatabase(path, options, status, error);
-#endif
+#else
```

Comment:
In commit "build: Enable -Wunreachable-code" (fa8adbe7c17b16cf7ecd16eb9f3f1792e9da2876)

Not a big deal, but IMO, this change is not an improvement. The unreachable code warning seems helpful to catch cases where code is intentionally unreachable. Or to encourage rewriting code where code is intentionally reachable but written in a hard to understand way.

But this code was intentionally unreachable in obvious way with two return statements right next to each other. It was written this way so changes to code could be checked at compile time without developers needing rebuild with --without-bdb and --with-sqlite=yes options. So this seems like a case where the compiler warning is unhelpful, and it would be nice if there was a way to disable it.